packages/website/blog/2022-12-01-asts-and-typescript-eslint.md

```diff
@@ -0,0 +1,118 @@
+---
+authors:
+  - image_url: https://www.joshuakgoldberg.com/img/josh.jpg
+    name: Josh Goldberg
+    title: TypeScript ESLint Maintainer
+    url: https://github.com/JoshuaKGoldberg
+description: Describing what an AST (Abstract Syntax Tree) is and why it's useful for ESLint and TypeScript tooling.
+slug: asts-and-typescript-eslint
+tags: [ast, abstract syntax tree, parser, parsing, prettier]
+title: ASTs and TypeScript-ESLint
+---
+
+Programmers who work with tools like [ESLint](https://eslint.org) and [Prettier](https://prettier.io) often refer to ASTs.
+But what is an AST, why is it useful for these kinds of tools, and how does that interact with ESLint and TypeScript tooling?
+Let's dig in!
+
+## What's an AST?
+
+_Static analysis_ tools are those that look at code without running it.
+They typically _parse_ code, or transform it from a string into a standard format they can reason about known as an **Abstract Syntax Tree** (AST).
+ASTs are called such because although they might contain information on the location of constructs within source code, they are an abstract representation that cares more about the semantic structure.
+
+> In other words, an AST is a description of your code's syntax.
+
+### An Example AST
+
+Take this single line of code:
+
+```js
+1 + 2;
+```
+
+ESLint's AST format, **[ESTree]**, would describe that line of code as an object like:
+
+```json
+{
+  "type": "ExpressionStatement",
+  "expression": {
+    "type": "BinaryExpression",
+    "left": {
+      "type": "Literal",
+      "value": 1,
+      "raw": "1"
+    },
+    "operator": "+",
+    "right": {
+      "type": "Literal",
+      "value": 2,
+      "raw": "2"
+    }
+  }
+}
+```
+
+Each piece of code described within an AST description is referred to as a **node**, or AST node.
+Each node is given a **node type** indicating the type of code syntax it represents
+That code snippet includes four nodes of the following types:
+
+- _ExpressionStatement_: `1 + 2;`
+- _BinaryExpression_: `1 + 2`
+- _Literal_: `1`
+- _Literal_: `2`
+
+That ESTree object representation of the code is what static analysis tools such as [ESLint](https://eslint.org) and [Prettier](https://prettier.io) work with.
+
+## AST Formats
+
+ESTree is more broadly used than just for ESLint -- it is the de facto community standard.
```

Comment:
```suggestion
ESTree is more broadly used than just for ESLint -- it is a popular community standard.
```

it's not really a de-facto standard; there are a lot of different ASTs that are used (eg typescript's own AST). But it is probably the most popular